packages/pwa/app/components/header/index.test.js

```diff
@@ -4,112 +4,24 @@
-import {setupServer} from 'msw/node'
-import {rest} from 'msw'
-import {
-    exampleTokenReponse,
-    mockedCustomerProductLists,
-    mockedRegisteredCustomer
-} from '../../commerce-api/mock-data'
-import {Route, Switch} from 'react-router-dom'
-import {createMemoryHistory} from 'history'
-jest.mock('../../commerce-api/utils', () => {
-    const originalModule = jest.requireActual('../../commerce-api/utils')
+
+jest.mock('@chakra-ui/react', () => {
+    const originalModule = jest.requireActual('@chakra-ui/react')
     return {
         ...originalModule,
-        isTokenValid: jest.fn().mockReturnValue(true)
+        useMediaQuery: jest.fn().mockReturnValue([true])
     }
 })
 
-const MockedComponent = ({history}) => {
-    const customer = useCustomer()
-    useEffect(() => {
-        if (!customer.isRegistered) {
-            customer.login('customer@test.com', 'password1')
-        }
-    }, [])
-    const onAccountClick = () => {
-        // Link to account page for registered customer, open auth modal otherwise
-        if (customer.isRegistered) {
-            history.push('/en-GB/account')
-        }
-    }
-    const onWishlistClick = () => {
-        history.push('/en-GB/account/wishlist')
-    }
-
-    return (
-        <Switch>
-            <Route path={'/'}>
-                <div>
-                    <Header onMyAccountClick={onAccountClick} onWishlistClick={onWishlistClick} />
-                    <div>home page</div>
-                </div>
-            </Route>
-        </Switch>
-    )
-}
-MockedComponent.propTypes = {
-    history: PropTypes.object
-}
-
-const server = setupServer(
-    rest.post('*/customers/actions/login', (req, res, ctx) =>
-        res(ctx.set('authorization', `Bearer guesttoken`), ctx.json(mockedRegisteredCustomer))
-    ),
-    rest.post('*/oauth2/authorize', (req, res, ctx) =>
-        res(ctx.delay(0), ctx.status(303), ctx.set('location', `/testcallback`))
-    ),
-    rest.get('*/oauth2/authorize', (req, res, ctx) =>
-        res(ctx.delay(0), ctx.status(303), ctx.set('location', `/testcallback`))
-    ),
-    rest.get('*/testcallback', (req, res, ctx) => {
-        return res(ctx.delay(0), ctx.status(200))
-    }),
-    rest.post('*/oauth2/login', (req, res, ctx) =>
-        res(ctx.delay(0), ctx.status(200), ctx.json(mockedRegisteredCustomer))
-    ),
-    rest.get('*/customers/:customerId', (req, res, ctx) =>
-        res(ctx.delay(0), ctx.status(200), ctx.json(mockedRegisteredCustomer))
-    ),
-    rest.post('*/oauth2/token', (req, res, ctx) =>
-        res(
-            ctx.delay(0),
-            ctx.json({
-                customer_id: 'test',
-                access_token: 'testtoken',
-                refresh_token: 'testrefeshtoken',
-                usid: 'testusid',
-                enc_user_id: 'testEncUserId'
-            })
-        )
-    ),
-    rest.get('*/oauth2/logout', (req, res, ctx) =>
-        res(ctx.delay(0), ctx.status(200), ctx.json(exampleTokenReponse))
-    )
```

Comment:
all of these is trying to mock `useCustomer.isRegister`!

we can just simply mock the that module 

```
jest.mock('../../commerce-api/hooks/useCustomer', () => {
    return () => ({
        isRegistered: true
    })
})
```

Reply:
I think the previous test was more thorough as we're simulating a real API interaction using Mock Service Worker https://mswjs.io/. 

If we just mock the module we're missing testing the API.

https://kentcdodds.com/blog/stop-mocking-fetch

Reply:
we talked about it on slack but wanna leave a record..

IMO it is wrong to test API when this file is all about component unit test. We shouldn't be testing anything beyond functionalities of this component.

The current tests is essentially more about testing `useCustomer` hook than the actual Header component.